src/main/java/mods/eln/integration/waila/GhostNodeWailaProvider.kt

```diff
@@ -0,0 +1,82 @@
+package mods.eln.integration.waila
+
+import cpw.mods.fml.common.Optional
+import mcp.mobius.waila.api.IWailaConfigHandler
+import mcp.mobius.waila.api.IWailaDataAccessor
+import mcp.mobius.waila.api.IWailaDataProvider
+import mods.eln.Eln
+import mods.eln.misc.Coordonate
+import mods.eln.packets.GhostNodeWailaRequestPacket
+import net.minecraft.client.Minecraft
+import net.minecraft.entity.player.EntityPlayerMP
+import net.minecraft.item.ItemStack
+import net.minecraft.nbt.NBTTagCompound
+import net.minecraft.tileentity.TileEntity
+import net.minecraft.world.World
+
+@Optional.Interface(iface = "mcp.mobius.waila.api.IWailaDataProvider", modid = "Waila")
+class GhostNodeWailaProvider(private val transparentNodeHandler: TransparentNodeHandler): IWailaDataProvider {
+    companion object {
+        var updateTime = Minecraft.getSystemTime()
+    }
+
+    private fun getGhostData(coord: Coordonate): GhostNodeWailaData? {
+        var ghostData: GhostNodeWailaData? = null
+
+        if (Minecraft.getSystemTime() - GhostNodeWailaProvider.updateTime > 2000) {
```

Comment:
The method get calls the method load of the cache, which does in our case only send the request packet to the server. So it is not really blocking, but still kind of a hack. That is why get throws an exception - the load method does not return the desired value.